cloud/blockstore/libs/storage/volume/volume_state.cpp

```diff
@@ -739,6 +741,13 @@ bool TVolumeState::CanAcceptClient(
     return newFillSeqNumber >= Meta.GetFillSeqNumber();
 }
 
+bool TVolumeState::ShouldForceTabletRestart(
+    const NProto::TVolumeClientInfo& info)
```

Comment:
So, we restart volume if either client id changed or mount seq number changed, or fill seq number changed.

No need to check fill generation because rw mount requests with wrong fill generation are be rejected. The case when disk is mounted with zero fill generation for the first time after filling is also satisfied: in this case, fill seq number of the request is zero, and Meta.GetFillSeqNumber is be positive, so the volume will restart.